chain/chain/src/types.rs

```diff
@@ -289,6 +289,12 @@ pub trait RuntimeAdapter: Send + Sync {
 
     fn get_flat_storage_state_for_shard(&self, shard_id: ShardId) -> Option<FlatStorageState>;
 
+    fn add_flat_storage_state_for_shard(
```

Comment:
Could you add a doc comment? This feels like an `.init` method which you want to call once, so I think it comes with some invariatns?